modules/react/tabs/lib/Tabs.tsx

```diff
@@ -19,29 +19,121 @@ export interface TabsProps {
   children: React.ReactNode;
 }
 
+/**
+ * `Tabs` is a container component that is responsible for creating a {@link TabsModel} and sharing
+ * it with its subcomponents using React context. It does not represent a real element.
+ *
+ * ```tsx
+ * <Tabs onSelect={data => console.log('Selected tab', data.id)}>
+ *   {child components}
+ * </Tabs>
+ * ```
+ *
+ * Alternatively, you may pass in a model using the hoisted model pattern.
+ *
+ * ```tsx
+ * const model = useTabsModel({
+ *   onSelect(data) {
+ *     console.log('Activated Tab', data.id);
+ *   },
+ * });
+ *
+ * <Tabs model={model}>{child components}</Tabs>
+ * ```
+ *
+ * See [Configuring a
+ * Model](/getting-started/for-developers/resources/compound-components/#configuring-a-model) for
+ * more information on model configuration.
+ */
 export const Tabs = createContainer()({
   displayName: 'Tabs',
   modelHook: useTabsModel,
   subComponents: {
+    /**
+     * `Tabs.List` is a `<div role="tablist">` element. It is a container for {@link TabsItem Tabs.Item}
+     * subcomponents.
+     *
+     * ```tsx
+     * <Tabs.List>{Tabs.Items}</Tabs.List>
+     * ```
+     */
     List: TabsList,
+    /**
+     * `Tabs.Item` is a `<button role="tab">` element. Each `Tabs.Item` is associated with a
+     * `Tabs.Panel` either by a `data-id` attribute or by the position in the list. If a `data-id`
+     * is provided, it must match the `data-id` passed to the corresponding `Tabs.Panel` component.
+     *
+     * ```tsx
+     * <Tabs.Item data-id="first">First Tab</Tabs.Item>
+     * ```
+     */
     Item: TabsItem,
+    /**
+     * `Tabs.Panel` is a `<div role="tabpanel">` element. It is focusable by default. Each
+     * `Tabs.Panel` is controlled by a `Tabs.Item` either by a `data-id` attribute or by the
+     * position in the list. If a `data-id` is provided, it must match the `data-id` passed to the
+     * corresponding `Tabs.Item` component.
+     *
+     * ```tsx
+     * <Tabs.Panel data-id="first">
+     *   {contents of tab panel}
+     * </Tabs.Panel>
+     * ```
+     */
     Panel: TabsPanel,
+    /**
+     * Optional subcomponent of tabs that makes dealing with dynamic tabs easier. `Tabs.Panels` only
+     * works if the model is given an array of items.
+     *
+     * ```tsx
+     * const items = [{ id: '1', name: 'first' }]
+     * <Tabs items={items}>
+     *   {tab list}
+     *   <Tabs.Panels>
+     *     {item => <Tabs.Panel>Contents of {item.name}</Tabs.Panel>}
+     *   </Tabs.Panels>
+     * </Tabs>
+     * ```
+     */
     Panels: TabsPanels,
+    /**
+     * The overflow button of a `Tabs.List`. When a tab list overflows, this overflow button is
+     * rendered. Overflow only works when the model is given an array of `items`.
+     *
+     * ## Usage
```

Comment:
This is nesting under an h3 and its sibling heading for "Props" is also an h4.

```suggestion
     * #### Usage
```

Reply:
I also noticed one of these on Pill and they seem to be inconsistently used. I think omitting them would be fine.

Reply:
There is code to normalize any heading levels inside a JSDoc comment, so the actual value doesn't matter. @alanbsmith suggested an h1, but that seemed a little hard. After talking with @jamesfan, we should remove all headings called "Usage" when inside the component API section.

Reply:
More context:

We do not consistently use "Usage" in the component API section. Some of the docs embedded examples in the usage section which is no longer allowed when using JSDoc. Instead codeblocks can be embedded in the component API. The component API section is meant to be an API doc and not a narrative. The narrative section is before the API section.